daemon/endpoint.go

```diff
@@ -151,7 +152,20 @@ func (h *putEndpointID) Handle(params PutEndpointIDParams) middleware.Responder
 
 	endpointmanager.Insert(ep)
 
-	return NewPutEndpointIDCreated()
+	add := labels.NewLabelsFromModel(params.Endpoint.Labels)
+
+	if len(add) > 0 {
+		endpointmanager.Mutex.Unlock()
+		errLabelsAdd := h.d.UpdateSecLabels(params.ID, add, labels.Labels{})
+		endpointmanager.Mutex.Lock()
+		if errLabelsAdd != nil {
+			log.Errorf("Could not add labels %v while creating an ep %s", add, params.ID)
+			return errLabelsAdd
```

Comment:
Small nit: why not return the error?
`return fmt.Errorf("Could not add labels %v while creating an ep %s: %s", add, params.ID, err)`